### PR TITLE
Remove unnecessary distinct

### PR DESCRIPTION
**Pull request by matllubos**

### CODE REVIEW

File: pyston/filters/managers.py [hunk 1]
```diff
@@ -174,6 +175,35 @@ def filter(self, resource, qs, request):
...
 
 
+def get_flat_lookups(q):
+    if isinstance(q, Q):
+        result = []
+        for child in q.children:
+            result += get_flat_lookups(child)
+        return result
+    else:
+        return [q[0]]
```
Comment (matllubos) [inline pyston/filters/managers.py:185]: return list of lookups from Q objects. For example:

user__issues__name = xy and user__first_name= ab

returns

[user__issues__name, user__first_name]

File: pyston/filters/managers.py [hunk 2]
```diff
@@ -174,6 +175,37 @@ def filter(self, resource, qs, request):
...
+
+
+def is_required_distinct_for_lookup(model, lookup, fail_first=False):
+    if '__' in lookup:
+        curr_lookup, next_lookup = lookup.split('__', 1)
+    else:
+        curr_lookup, next_lookup = lookup, None
+    try:
+        field = model._meta.get_field(curr_lookup)
+        if not field.is_relation:
```
Comment (matllubos) [inline pyston/filters/managers.py:195]: no relation no duplications

File: pyston/filters/managers.py [hunk 3]
```diff
@@ -174,6 +175,37 @@ def filter(self, resource, qs, request):
...
+            return False
+        elif field.many_to_many or field.one_to_many:
+            # m2m or o2m relations can cause duplications
```
Comment (matllubos) [inline pyston/filters/managers.py:198]: o2m or m2m duplications

File: pyston/filters/managers.py [hunk 4]
```diff
@@ -174,6 +175,37 @@ def filter(self, resource, qs, request):
...
+            return True
+        elif not next_lookup:
+            return False
```
Comment (matllubos) [inline pyston/filters/managers.py:201]: no next lookup no m2m or o2m means no duplications

File: pyston/filters/managers.py [hunk 5]
```diff
@@ -174,6 +175,37 @@ def filter(self, resource, qs, request):
...
+        else:
```
Comment (matllubos) [inline pyston/filters/managers.py:202]: next lookup can conain m2m or o2m relations

File: pyston/filters/managers.py [hunk 6]
```diff
@@ -174,6 +175,37 @@ def filter(self, resource, qs, request):
...
+            return is_required_distinct_for_lookup(field.related_model, next_lookup, True)
+    except FieldDoesNotExist:
+        # For annotations is distinct automatically required
+        return fail_first
```
Comment (matllubos) [inline pyston/filters/managers.py:206]: if field was not found there is two posibilities:
* it is some keyword like icontains, excact (first_name__contains) - no duplications
* it is lookup for annotation, there can be duplications and annotations should be find in the first call of this methods (cannot be used name issues__anotation only annotation)

File: pyston/filters/managers.py [hunk 7]
```diff
@@ -215,15 +247,26 @@ def _convert_logical_conditions(self, condition, resource, request):
                     mark_safe(ugettext('Invalid operator of condition "{}"').format(condition.source))
                 )
 
+    def _is_required_distinct(self, qs, q):
+        for lookup in get_flat_lookups(q):
+            if is_required_distinct_for_lookup(qs.model, lookup):
```
Comment (matllubos) [inline pyston/filters/managers.py:252]: if some lookup requires distinct then distict must be used

File: pyston/filters/managers.py [hunk 8]
```diff
@@ -215,15 +247,26 @@ def _convert_logical_conditions(self, condition, resource, request):
...
     def filter(self, resource, qs, request):
         try:
             parsed_conditions = self.parser.parse(request)
-            return (
-                qs.filter(pk__in=qs.filter(
-                    self._convert_logical_conditions(parsed_conditions, resource, request)).values('pk')
+            if parsed_conditions:
+                q = self._convert_logical_conditions(parsed_conditions, resource, request)
+                filtered_qs = qs.filter(
+                    self._convert_logical_conditions(parsed_conditions, resource, request)
```
Comment (Formulka) [inline pyston/filters/managers.py:262]: `filtered_qs = qs.filter(q)` ?

**Pull request merged**